kafkabp/consumer.go

```diff
@@ -43,13 +43,15 @@ type ConsumeMessageFunc func(ctx context.Context, msg *sarama.ConsumerMessage)
 type ConsumeErrorFunc func(err error)
 
 // Consumer defines the interface of a consumer struct.
+//
+// It's also a superset of (implements) baseplate.Healthyer.
 type Consumer interface {
 	io.Closer
 
 	Consume(ConsumeMessageFunc, ConsumeErrorFunc) error
 
 	// IsHealthy returns false after Consume returns.
-	IsHealthy() bool
+	IsHealthy(ctx context.Context) bool
```

Comment:
Should this just embed the `Healthyer` interface instead?

Reply:
There are two reasons:

1. Avoid the dependency
2. Since all the consumer implementations are unexported, this is where we document the behavior of the `IsHealthy` function, so changing this to embedding will make the document less useful, as people would need to jump to that definition to see how it actually looks like (it's not a problem for `io.Closer` as that one is "well-known").